include/deal.II/base/smartpointer.h

```diff
@@ -198,6 +198,11 @@ class SmartPointer
    * The identification for the subscriptor.
    */
   const char *const id;
+
+  /**
+   * The Smartpointer is invalidated when the object pointed to is destroyed.
+   */
+  bool valid = false;
```

Comment:
put the initialization into the constructor as we usually do -- I find this syntax confusing because it looks like a `const` variable with an initializer, but that's not what it actually is.